packages/cubejs-server-core/src/core/OrchestratorApi.ts

```diff
@@ -230,6 +231,33 @@ export class OrchestratorApi {
     }
   }
 
+  public async fetchSchema(dataSource: string, securityContext?: { [key: string]: any; }, renew?: boolean) {
+    const cacheDriver = this.orchestrator
+      .getQueryCache()
+      .getCacheDriver();
+
+    const cacheHash = crypto
+      .createHash('md5')
+      .update(JSON.stringify([
```

Comment:
It's a wrong cache key. You need to use `this.orchestrator.getQueryCache().getKey('FETCH_SCHEMA', getCacheHash({ dataSource, securityContext }))`